mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,384 @@
+
+.. code-block:: HTML
+
+	<a href="mailto:Support@example.com">Support@example.com</a>
+
+That is, an anchor tag that makes this an email link. You might want to do this if you are writing a blog engine and need it to send email for marketing, support, and other contacts, all to the same domain.
+
+#. Add the following ``EmailTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/z1EmailTagHelperCopy.cs
+   :language: c#
+   
+Notes: 
+
+- ASP.NET MVC 6 uses a naming convention that targets elements of the root class name (minus the *TagHelper* portion of the class name). In this example, the root name of **Email**\TagHelper is *email*, so the ``<email>`` tag will be targeted. This naming convention should work for most tag helpers and later on I'll show how to override it.
+
+- The ``EmailTagHelper`` class derives from `TagHelper <https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>`__. The ``TagHelper`` class provides the rich methods and properties we will examine in this tutorial.
+- The override ``Process`` method controls what the tag helper does when executed.  The ``TagHelper`` class also provides an asynchronous version (``ProcessAsync``) with the same parameters.  
+- The context parameter to ``Process`` (and ``ProcessAsync``) contains information associated with the execution of the current HTML tag. We will use the context parameter to get the content of our target element.
+- The output parameter to ``Process`` (and ``ProcessAsync``) contains a stateful HTML element representative of the original source used to generate an HTML tag and content.
+- Our class name has a suffix of **TagHelper**, which is required.
+
+.. The ``TargetElement`` attribute identifies "email" as the element to target for processing. If you don't specify a ``TargetElement`` attribute, ASP.NET MVC 6 uses a naming convention that targets elements of the root class name (minus the *TagHelper* portion of the class name). You could comment out this attribute (because the class is named **Email**\TagHelper), and it would target email tags. Using the attribute does make the intention explicit (that is, we are targeting **<email>** tags).
+
+2. To make the ``EmailTagHelper`` class available to all our Razor views, we will add the ``addTagHelper`` directive to the *Views/_ViewImports.cshtml* file:  
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/Views/_ViewImports.cshtml
+   :language: aspx-cs
+   :lines: 1-3
+   :emphasize-lines: 3
+
+To add a tag helper to a view, you first add the fully qualified name (``AuthoringTagHelpers.TagHelpers.EmailTagHelper``), and then the DLL name (*AuthoringTagHelpers*). You typically add tag helpers to the *Views/_ViewImports.cshtml* file as we've done here, but it's more common to use wildcard syntax to bring in all tag helpers with one line, as the following shows:
```

Comment:
> but it's more common to use wildcard syntax
Reverse the order of this bit with the wild card bit. Aka start by showing wild card version and have the user do that; then say if you want to be more explicit you can provide the full name.

This way, users will use the best/more common practice of using a wild card `@addTagHelper`.